src/main/java/org/launchcode/water_garden_tour/controllers/HomeController.java

```diff
@@ -1,24 +1,56 @@
 package org.launchcode.water_garden_tour.controllers;
 
 import org.launchcode.water_garden_tour.models.data.FeatureRepository;
+import org.launchcode.water_garden_tour.models.data.GardenRepository;
+import org.launchcode.water_garden_tour.models.garden.Garden;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
 @Controller
 public class HomeController {
 
     @Autowired
     private FeatureRepository featureRepository;
 
+    @Autowired
+    private GardenRepository gardenRepository;
+
     @GetMapping("")
     public String index(Model model) {
+        List<Garden> featuredGardens = new ArrayList<>();
+        List<Garden> allGardens = gardenRepository.findAll();
+        Integer[] allGardenIds = new Integer[allGardens.size()];
+        int curIndex = 0;
+
+        for (Garden garden : allGardens) {
+            int curGardenId = garden.getId();
+            allGardenIds[curIndex] = curGardenId;
+            curIndex++;
+        }
+
+        do {
+            double randomNumberDbl = Math.random() * allGardenIds.length;
+            int randomNumber = (int) Math.floor(randomNumberDbl);
+
+            int curGardenId = allGardenIds[randomNumber];
+            Optional<Garden> optGarden = gardenRepository.findById(curGardenId);
+            Garden curGarden = optGarden.get();
+
+            if (!featuredGardens.contains(curGarden)) {
+                featuredGardens.add(curGarden);
+            }
+        } while (featuredGardens.size() < 3);
```

Comment:
Thanks! Great improvement, that looks much cleaner! I put the gardens into an array thinking it would be easier to use with a random number generator but it was totally unnecessary.